R/RLearner_classif_mxff.R

```diff
@@ -0,0 +1,416 @@
+#' @export
+makeRLearner.classif.mxff = function() {
+  makeRLearnerClassif(
+    cl = "classif.mxff",
+    package = "mxnet",
+    par.set = makeParamSet(
+      # architectural hyperparameters
+      makeIntegerLearnerParam(id = "layers", lower = 1L, upper = 3L, default = 1L),
+      makeIntegerLearnerParam(id = "num.layer1", lower = 1L, default = 1L),
+      makeIntegerLearnerParam(id = "num.layer2", lower = 1L, default = 1L,
+        requires = quote(layers > 1)),
+      makeIntegerLearnerParam(id = "num.layer3", lower = 1L, default = 1L,
+        requires = quote(layers > 2)),
+      makeDiscreteLearnerParam(id = "act1", default = "tanh",
+        values = c("tanh", "relu", "sigmoid", "softrelu")),
+      makeDiscreteLearnerParam(id = "act2", default = "tanh",
+        values = c("tanh", "relu", "sigmoid", "softrelu"),
+        requires = quote(layers > 1)),
+      makeDiscreteLearnerParam(id = "act3", default = "tanh",
+        values = c("tanh", "relu", "sigmoid", "softrelu"),
+        requires = quote(layers > 2)),
+      makeDiscreteLearnerParam(id = "act.out", default = "softmax",
+        values = c("rmse", "softmax", "logistic")),
+      makeLogicalLearnerParam(id = "conv.layer1", default = FALSE),
+      makeLogicalLearnerParam(id = "conv.layer2", default = FALSE,
+        requires = quote(layers > 1 && conv.layer1 == TRUE)),
+      makeLogicalLearnerParam(id = "conv.layer3", default = FALSE,
+        requires = quote(layers > 2 && conv.layer2 == TRUE)),
+      makeIntegerVectorLearnerParam(id = "conv.data.shape",
+        requires = quote(conv.layer1 == TRUE)),
+      makeIntegerLearnerParam(id = "conv.kernel11", lower = 1L,
+        requires = quote(conv.layer1 == TRUE)),
+      makeIntegerLearnerParam(id = "conv.kernel12", lower = 1L,
+        requires = quote(conv.layer1 == TRUE)),
```

Comment:
Yes, we talked about that and you're completely correct with the generate design function. Although in mlr tuning vector params **of fixed length** works. 

See for example with H2O:

```
library(mlr)
lrn = makeLearner('classif.h2o.deeplearning')
ps = makeParamSet(
   makeIntegerVectorParam('hidden', lower = c(1,1), upper = c(5,5), len = 2)

ctrl = makeTuneControlRandom(maxit = 1)
tuneParams(lrn, iris.task, hout, mmce, ps, ctrl)
 
```